File: yw/dom/DocumentFragment.lua
--[[
    Copyright (c) 2024, Oh Inseo (YJK) <dhdlstjtr@gmail.com>
    SPDX-License-Identifier: BSD-3-Clause
    This software may contain third-party material. For more info, see README.
]]
local Node   = require "yw.dom.Node"
local object = require "yw.common.object"


---https://dom.spec.whatwg.org/#documentfragment
---@class DOM_DocumentFragment : DOM_Node
---@field host DOM_Node
local DocumentFragment = object.create(Node)

---@param document DOM_Document
---@return DOM_DocumentFragment
function DocumentFragment:new(document)
    local o = Node.new(self, document) --[[@as DOM_DocumentFragment]]
    o.isDocumentFragment = true
    o.host = nil
    return o
end

return DocumentFragment

File: dom/document_fragment.go
// This file is part of YW project. Copyright 2025 Oh Inseo (YJK)
// SPDX-License-Identifier: BSD-3-Clause
// See LICENSE for details, and LICENSE_WHATWG_SPECS for WHATWG license information.

package dom

// DocumentFragment represents a [DOM DocumentFragment].
//
// [DOM DocumentFragment]: https://dom.spec.whatwg.org/#documentfragment
type DocumentFragment interface {
	Node

	// Host returns [host] of the DocumentFragment.
	//
	// [host]: https://dom.spec.whatwg.org/#concept-documentfragment-host
	Host() Node
}
